CodeAssignment_MVVM_Combine/Sources/Controllers/HomeViewController.swift

```diff
@@ -17,63 +18,76 @@ class HomeViewController: UIViewController {
         button.titleLabel?.font = UIFont.boldSystemFont(ofSize: 14)
         button.setTitle("GET", for: .normal)
         button.setTitleColor(UIColor.systemBlue, for: .normal)
-        /// Snapkitライブラリで既にtranslatesAutoresizingMaskIntoConstraintsをfalseにしているので、記載不要
+        // Snapkitライブラリで既にtranslatesAutoresizingMaskIntoConstraintsをfalseにしているので、記載不要
         button.clipsToBounds = true
         button.layer.cornerRadius = 8
         button.layer.borderWidth = 1
         button.layer.borderColor = UIColor.systemGray.cgColor
-        /// addActionを使うことで、objc funcを使わなくても済む
-        /// UIActionのclosureを用いて関数を呼び出す
-        button.addAction(UIAction { [weak self] _ in
+        // addActionを使うことで、objc funcを使わなくても済む
+        // UIActionのclosureを用いて関数を呼び出す
+        button.addAction(.init { [weak self] _ in
             self?.didTapGetAPIButton()
         }, for: .touchUpInside)
         
         return button
     }()
     
-
+    /// ViewModel
+    private let viewModel = HomeViewModel()
+    /** storeでAnyCancellableを保存しておいて、当該の変数がdeinitされるとき、subscribeをキャンセルする方法
+    - Setで複数のSubscription（購読）を１つにまとめることができ、Subscriptionの値を保持する
+     */
+    private var cancellables = Set<AnyCancellable>()
+    private var repositories = [Repository]()
+    
     override func viewDidLoad() {
         super.viewDidLoad()
         
-        setAddSubViews()
-        setupController()
-        setupConstraints()
         setupUI()
+        bind()
     }
 }
 
-/// Functions & Logics
+// MARK: - Functions & Logics
 extension HomeViewController {
-    /// Controllerをセットアップする
-    func setupController() {
+    /// ViewControllerのUIをセットアップする
+    private func setupUI() {
         view.backgroundColor = .systemBackground
-        
-        
-    }
-    
-    /// カスタムで作ったViewを全部追加する
-    func setAddSubViews() {
-        view.addSubview(getAPIButton)
+        setAddSubViews()
+        setupConstraints()
     }
     
-    /// UIをセットアップする
-    func setupUI() {
-        setupGetAPIButton()
+    /** ViewModelなどViewController側で常に監視しておくべき対象を、セットアップ
+     イベント発生時に正常にデータバインドをするために、Observerを設定する感じ
+     
+    - Combineで流れたきたデータのアウトプットsinkする
+    - sink : Publisherからのイベントを購読する.  つまり、イベントを受信したときの処理を指定できる。
+    - receive(on:)：イベントを受け取るスレッドを指定する
+    - store: cancellabeなどを保
+     */
+    private func bind() {
+        viewModel.repositoriesSubject
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] repositories in
+                guard let self = self else { return }
```

Comment:
```suggestion
                guard let self else { return }
```
Swift 5.7から↑のように書けるようになってます
https://github.com/apple/swift-evolution/blob/main/proposals/0345-if-let-shorthand.md